### PR TITLE
Update GetPolicies to mirror NiFi 1.12.1 InvokeHttp

### LINKED ISSUE #45: Improve SSL Context support in GetPolicies
This same concern may apply to other processors initializing SSL Context in the same way.

There is not adequate support for a Key Password indepdent of the Keystore Password. The NiFi interface allows for specifying both a Keystore Password and a Key Password as shown here
Screenshot from 2020-12-11 19-39-33

The current implementation in GetPolicies only takes into account the Keystore Password when initializing the key manager
https://github.com/greymatter-io/nifi-sdk/blob/main/gmd-sdk/nifi-data-processors/src/main/scala/com/deciphernow/greymatter/data/nifi/processors/GetPolicies.java#L288-L303

The description for Key Password is as follows: `The password for the key. If this is not specified, but the Keystore Filename, Password, and Type are specified, then the Keystore Password will be assumed to be the same as the Key Password.`

### PR DESCRIPTION
**Pull request by dborncamp**
- Also update test per change in order of returned things from ACM Server
- Update the circle docker-compose to reference NiFi 1.12.1

Relates to #45

### CODE REVIEW

File: gmd-sdk/nifi-data-processors/src/main/scala/com/deciphernow/greymatter/data/nifi/processors/GetPolicies.java [hunk 1]
```diff
@@ -257,84 +249,17 @@ public void setUpClient(final ProcessContext context) throws IOException, Unreco
...
 
+        // Apply the TLS configuration if present
         final SSLContextService sslService = context.getProperty(PROP_SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
-        final SSLContext sslContext = sslService == null ? null : sslService.createSSLContext(ClientAuth.NONE);
-
-        // check if the ssl context is set and add the factory if so
-        if (sslContext != null) {
-            setSslSocketFactory(okHttpClientBuilder, sslService, sslContext);
+        if (sslService != null) {
+            final TlsConfiguration tlsConfiguration = sslService.createTlsConfiguration();
```
Comment (lucasmoten) [inline gmd-sdk/nifi-data-processors/src/main/scala/com/deciphernow/greymatter/data/nifi/processors/GetPolicies.java:255]: Just a note: I was able to tracedown where TlsConfiguration was added to nifi in this commit: https://github.com/apache/nifi/commit/441781cec50f77d9f1e65093f55bbd614b8c5ec6

File: gmd-sdk/nifi-data-processors/src/main/scala/com/deciphernow/greymatter/data/nifi/processors/GetPolicies.java [hunk 2]
```diff
@@ -257,84 +249,17 @@ public void setUpClient(final ProcessContext context) throws IOException, Unreco
...
+            OkHttpClientUtils.applyTlsToOkHttpClientBuilder(tlsConfiguration, okHttpClientBuilder);
```
Comment (lucasmoten) [inline gmd-sdk/nifi-data-processors/src/main/scala/com/deciphernow/greymatter/data/nifi/processors/GetPolicies.java:256]: In same PR the OkHttpClientUtils https://github.com/apache/nifi/commit/441781cec50f77d9f1e65093f55bbd614b8c5ec6#diff-7251dc1adc04dc044eec2b217c8e3b70d84b6331b153f98f8520aa39829ea93c

Comment (lucasmoten) [reply gmd-sdk/nifi-data-processors/src/main/scala/com/deciphernow/greymatter/data/nifi/processors/GetPolicies.java:256]: It wraps this 
```
final SSLSocketFactory sslSocketFactory = SslContextFactory.createSSLSocketFactory(tlsConfiguration);
```

**Review by lucasmoten — COMMENTED** [review]
This is a step in the right direction, but unfortunately doesn't seem to actually accomodate a different key passwrod from keystore password.  I'm trying to find out if that is even supported in NiFi anywhere as its implied but the implementations differed across processors and thats why NIFI-7407 intended to streamline some of that.

File: gmd-sdk/nifi-data-processors/src/main/scala/com/deciphernow/greymatter/data/nifi/processors/GetPolicies.java (hunk 2 above)
Comment (lucasmoten) [reply gmd-sdk/nifi-data-processors/src/main/scala/com/deciphernow/greymatter/data/nifi/processors/GetPolicies.java:256]: It should be initializing the keystore with keystorepassword + keypassword or just keystorepassword here https://github.com/apache/nifi/commit/441781cec50f77d9f1e65093f55bbd614b8c5ec6#diff-2876b4f0115a43570c21c95e516760187b37ea5db3990ea0a12f9056ccf01a8eR133

**Review by lucasmoten — APPROVED** [review]
Approving this as it brings in line with InvokeHttp and the refactored SSL handling across NiFi, but we should keep issue #45 open until that is verified as resolved.  From my testing, it wasnt picking up differences in keystore password vs key password and would fail with the  java.security.UnrecoverableKeyException: Cannot recover key: java.security.UnrecoverableKeyException: Cannot recover key

File: gmd-sdk/nifi-data-processors/src/test/scala/com/deciphernow/greymatter/data/nifi/processors/GetPoliciesTest.java [hunk 3]
```diff
@@ -93,7 +93,7 @@ public void testPoliciesProcessor() {
         assertEquals("", flowFile.getAttribute("getpolicies.response.body"));  // it is 200, no errors should be here
         assertEquals("https://127.0.0.1:8081/convert/addpermissions", flowFile.getAttribute("getpolicies.request.url"));
         assertEquals("CN=gm-data,OU=Engineering,O=Untrusted Example,L=Baltimore,ST=MD,C=US", flowFile.getAttribute("getpolicies.remote.dn"));
-        assertEquals("{\"background\":\"007A33\",\"label\":\"UNCLASSIFIED\",\"foreground\":\"#FFFFFF\"}", flowFile.getAttribute("gmdata.security"));
+        assertEquals("{label=UNCLASSIFIED, foreground=#FFFFFF, background=#007A33}", flowFile.getAttribute("gmdata.security"));
```
[the remark this answers is not in the document]
Comment (dborncamp) [reply gmd-sdk/nifi-data-processors/src/test/scala/com/deciphernow/greymatter/data/nifi/processors/GetPoliciesTest.java:96]: That's what the new version (0.3.3) now returns so I think so.

**Pull request merged**